File: 198-house-robber/198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        
        case1, case2 = 0, 0
        
        for i in range(len(nums)):
            
            if(i % 2 == 0):
                case1 = max(case2, nums[i] + case1)  
            else:
                case2 = max(case1, nums[i] + case2)
                
        return max(case1, case2)
        

File: 198-house-robber/198-house-robber.c
# define MAX(a,b) (a>b?a:b)

int rob(int* nums, int numsSize){
    
    int rob_case1 = 0, rob_case2 = 0;
    
    
    for(int i = 0; i < numsSize; i++){
        
        if(i % 2 == 0)
            rob_case1 = MAX(rob_case2, rob_case1 + nums[i]);
        else
            rob_case2 = MAX(rob_case1, rob_case2 + nums[i]);
        
    }
    
    return MAX(rob_case1, rob_case2);
}